pyproject.toml

```diff
@@ -77,6 +77,7 @@ rank-bm25 = {optional = true, version = "^0.2.2"}
 scikit-learn = {optional = true, version = "*"}
 spacy = {optional = true, version = "^3.7.1"}
 aiohttp = "^3.8.6"
+redis = "^5.0.1"
```

Comment:
let's avoid adding reddis as a direct dependency. I see you are already doing try/except imports 👍🏻